addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6039,6 +6039,22 @@ registry.SnippetMove = SnippetOptionWidget.extend(ColumnLayoutMixin, {
     },
 });
 
+registry.AlignImage = SnippetOptionWidget.extend({
+    selectClass: function (previewMode, widgetValue, params) {
+        // if we are changing the alignment
+        // (alignment select has data-state-to-first-class="true") and
+        // if the link was created after this fix, `a` isn't `display: contents`
+        // we set its display to contents when aligning the image.
+        if (params.stateToFirstClass === "true") {
+            const node = this.$target.get(0);
+            if (node.nodeName === "IMG" && node.parentNode?.nodeName === "A") {
+                node.parentNode.style.display = "contents";
```

Comment:
I really don't think this is a safe solution, as I said here: https://github.com/odoo/odoo/pull/221524#discussion_r2269314463 -> display:contents is known to not properly work everywhere. But maybe I am wrong I'll let someone from the team handle this one from. I'll create a task about and someone will pick it up.

I still think there should be a better way to handle this.